torchbenchmark/score/compute_score.py

```diff
@@ -7,32 +7,137 @@
 import math
 import sys
 import os
+import re
 import yaml
+import importlib
 
 from tabulate import tabulate
+from pathlib import Path
+from collections import defaultdict
 
-SPEC_FILE_DEFAULT = "score.yml"
+SPEC_FILE_DEFAULT = "torchbenchmark/score/score.yml"
 TARGET_SCORE_DEFAULT = 1000
+TORCHBENCH_V0_REF_DATA = "torchbenchmark/score/torchbench_0.0.yaml"
+TORCHBENCH_V1_REF_DATA = "" # Placeholder for v1 score
 
-def compute_score(config, data, fake_data=None):
-    target = config['target']
-    score = 0.0
-    weight_sum = 0.0
-    for name in config['benchmarks']:
-        cfg = config['benchmarks'][name]
-        weight, norm = cfg['weight'], cfg['norm']
-        weight_sum += weight
-        measured_mean = [b['stats']['mean'] for b in data['benchmarks'] if b['name'] == name]
-        assert len(measured_mean) == 1, f"Missing data for {name}, unable to compute score"
-        measured_mean = measured_mean[0]
-        if fake_data is not None and name in fake_data:
-            # used for sanity checks on the sensitivity of the score metric
-            measured_mean = fake_data[name]
-        benchmark_score = weight * math.log(norm / measured_mean)
-        # print(f"{name}: {benchmark_score}")
-        score += benchmark_score
-
-    score = target * math.exp(score)
-    assert abs(weight_sum - 1.0) < 1e-6, f"Bad configuration, weights don't sum to 1, but {weight_sum}"
-    return score
+def _get_model_task(model_name):
+    """
+    Helper function which extracts the task the model belongs to
+    by iterating over the Model attributes.
+    """
+    MODEL_PATH="../../models"
+    p = Path(__file__).parent.joinpath(MODEL_PATH + model_name)
+    if(p):
+        module = importlib.import_module(f'torchbenchmark.models.{model_name}', package=__name__)
+        Model = getattr(module, 'Model')
+    return Model.task.value
 
+class TorchBenchScore:
+    def __init__(self, spec=SPEC_FILE_DEFAULT, target=TARGET_SCORE_DEFAULT, ref_data=None):
```

Comment:
I think ref_data should be a required argument, at least in the current design.
a) not clear what the point of constructing a score without a ref data is
b) you don't provide a way to add ref_data later anyway

given these, i would say it's simpler/cleaner to just require it done in _init_ and then delete the if/else checks in other parts of code.  Once you construct the score, it's ready to use- easy to reason about that way.

Reply:
Fixed